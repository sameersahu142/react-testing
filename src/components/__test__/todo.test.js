import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Todo from "../todo";

afterEach(() => {
  cleanup();
});

test("should render not completed todo component", () => {
  const todo = { id: 1, title: "Make Lunch", completed: false };
  render(<Todo todo={todo} />);
  const todoElement = screen.getByTestId("todo-1");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("Make Lunch");
  expect(todoElement).not.toContainHTML('<strike>');
});

test("should render completed todo component", () => {
  const todo = { id: 2, title: "Make dinner", completed: true };
  render(<Todo todo={todo} />);
  const todoElement = screen.getByTestId("todo-2");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("Make dinner");
  expect(todoElement).toContainHTML('</strike>');
});

test("matches snapshot", () => {
  const todo = { id: 1, title: "Make Lunch", completed: false };
  const tree = renderer.create(<Todo todo={todo} />).toJSON();
  expect(tree).toMatchSnapshot();
})
