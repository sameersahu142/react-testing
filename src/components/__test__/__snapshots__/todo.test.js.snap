// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <h1>
    Make Lunch
  </h1>
</div>
`;
